source/modulo_new_core/src/translators/WriteStateConversion.cpp

```diff
@@ -1,30 +1,65 @@
-#include "modulo_core/communication/message_passing/WriteStateConversion.hpp"
+#include "modulo_new_core/translators/WriteStateConversion.hpp"
 
-#include <state_representation/exceptions/IncompatibleReferenceFramesException.hpp>
+#include <std_msgs/msg/bool.hpp>
+#include <std_msgs/msg/float64.hpp>
+#include <std_msgs/msg/float64_multi_array.hpp>
+#include <std_msgs/msg/string.hpp>
 
 using namespace state_representation::exceptions;
 
-namespace modulo::core::communication::state_conversion {
+namespace modulo_new_core::translators {
+
+static void write_point(geometry_msgs::msg::Point& msg, const Eigen::Vector3d& vector) {
+  msg.x = vector.x();
+  msg.y = vector.y();
+  msg.z = vector.z();
+}
+
+static void write_vector3(geometry_msgs::msg::Vector3& msg, const Eigen::Vector3d& vector) {
+  msg.x = vector.x();
+  msg.y = vector.y();
+  msg.z = vector.z();
+}
+
+static void write_quaternion(geometry_msgs::msg::Quaternion& msg, const Eigen::Quaterniond& quat) {
+  msg.w = quat.w();
+  msg.x = quat.x();
+  msg.y = quat.y();
+  msg.z = quat.z();
+}
+
+void write_msg(geometry_msgs::msg::Point& msg, const state_representation::CartesianState& state, const rclcpp::Time&) {
+  if (state.is_empty()) throw EmptyStateException(state.get_name() + " state is empty while attempting to publish it");
+  write_point(msg, state.get_position());
+}
+
+void write_msg(geometry_msgs::msg::Vector3& msg, const state_representation::CartesianState& state, const rclcpp::Time&) {
+  if (state.is_empty()) throw EmptyStateException(state.get_name() + " state is empty while attempting to publish it");
+  write_vector3(msg, state.get_position());
+}
+
 void write_msg(geometry_msgs::msg::Quaternion& msg, const state_representation::CartesianState& state, const rclcpp::Time&) {
   if (state.is_empty()) throw EmptyStateException(state.get_name() + " state is empty while attempting to publish it");
-  // orientation
-  msg.w = state.get_orientation().w();
-  msg.x = state.get_orientation().x();
-  msg.y = state.get_orientation().y();
-  msg.z = state.get_orientation().z();
+  write_quaternion(msg, state.get_orientation());
 }
 
+void write_msg(geometry_msgs::msg::Accel& msg, const state_representation::CartesianState& state, const rclcpp::Time&) {
+  if (state.is_empty()) throw EmptyStateException(state.get_name() + " state is empty while attempting to publish it");
+  write_vector3(msg.linear, state.get_linear_acceleration());
+  write_vector3(msg.angular, state.get_angular_acceleration());
+}
+
+void write_msg(geometry_msgs::msg::AccelStamped& msg, const state_representation::CartesianState& state, const rclcpp::Time& time) {
+  msg.header.stamp = time;
+  msg.header.frame_id = state.get_reference_frame();
+  write_msg(msg.accel, state, time);
```

Comment:
You should invoke the other method first, since it checks for emptiness. Otherwise you only partially mutate the `msg`, for example with a potentially unexpected reference frame. Better to throw early without modifying the output argument.
```suggestion
  write_msg(msg.accel, state, time);
  msg.header.stamp = time;
  msg.header.frame_id = state.get_reference_frame();
```
I see it was already this way for the other stamped functions in this file, so you were just following the previous template. However I would encourage you to refactor the other functions to improve the behaviour in this manner.